app/controllers/api/base_controller.rb

```diff
@@ -27,7 +27,8 @@ def enqueue_web_hook_jobs(version)
 
   def verify_with_otp
     otp = request.headers["HTTP_OTP"]
-    return if @api_key.user.mfa_api_authorized?(otp)
+    return unless @api_key.mfa_enabled?
+    return if @api_key.user.otp_verified?(otp)
```

Comment:
IMO, it would be better if created a new function in api_key to check mfa_enabled and otp_verified? in the same function.

Reply:
Please add a functional test for this logic as well.

Reply:
Makes sense, I added the `mfa_authorized?` function in `api_key` and added tests to controllers that use the `verify_with_otp` before action (deletion, rubygems and owners)